salt/knox/init.sls

```diff
@@ -5,15 +5,12 @@
 {% set pnda_mirror = pillar['pnda_mirror']['base_url'] %}
 {% set misc_packages_path = pillar['pnda_mirror']['misc_packages_path'] %}
 {% set mirror_location = pnda_mirror + misc_packages_path %}
-{% set namenode_host = salt['pnda.get_hosts_by_hadoop_role']('HDFS', 'NAMENODE')[0] %}
-{% set hive_host = salt['pnda.get_hosts_by_hadoop_role']('HIVE', 'HIVE_SERVER')[0] %}
 {% set webhdfs_host = salt['pnda.get_hosts_by_hadoop_node']('MGR01')[0] %}
 {% set hbase_rest_host = salt['pnda.get_hosts_by_hadoop_node']('MGR01')[0] %}
+{% set namenode_host = salt['pnda.get_hosts_by_hadoop_role']('HDFS', 'NAMENODE')[0] %}
 {% set yarn_rm_hosts = salt['pnda.get_hosts_by_hadoop_role']('YARN', 'RESOURCEMANAGER') %}
 {% set yarn_ha_enabled = (yarn_rm_hosts is not none and yarn_rm_hosts|length>1) %}
-{% set mr2_history_server_host = salt['pnda.get_hosts_by_hadoop_role']('MAPREDUCE2', 'HISTORYSERVER')[0] %}
-{% set spark_history_server_host = salt['pnda.get_hosts_by_hadoop_role']('SPARK', 'SPARK_JOBHISTORYSERVER')[0] %}
-{% set ambari_server_host = salt['pnda.get_hosts_for_role']('hadoop_manager')[0] %}
+{% set hadoop_manager_host = salt['pnda.get_hosts_for_role']('hadoop_manager')[0] %}
```

Comment:
This change is incorrect. This isn't a generic Hadoop manager, it's Ambari. The knox config for Ambari will not work for anything else.